public/app/plugins/panel/barchart/bars.ts

```diff
@@ -286,7 +286,14 @@ export function getConfig(opts: BarsOptions, theme: GrafanaTheme2) {
     : {};
 
   let barsBuilder = uPlot.paths.bars!({
-    radius: barRadius,
+    radius: pctStacked
+      ? 0
+      : !isStacked
+      ? barRadius
+      : (u: uPlot, seriesIdx: number) => {
+          let isTopmostSeries = seriesIdx === u.data.length - 1;
+          return isTopmostSeries ? [barRadius, 0] : [0, 0];
+        },
```

Comment:
This isn't the most readable logic 🤔 could we maybe break this out as an util func or even setting a value for `radius` above line 288 using if branches? (not as concise but maybe more readable option?)

Reply:
i guess it's in the eye of the beholder? i find this to be much more difficult to read with a soup of braces and packed if/else one-liners:

```ts
let finalBarRadius: uPlot.Series.BarsPathBuilderRadius;

if (pctStacked) {
  finalBarRadius = 0;
} else if (!isStacked) {
  finalBarRadius = barRadius;
} else {
  finalBarRadius = (u: uPlot, seriesIdx: number) => {
     let isTopmostSeries = seriesIdx === u.data.length - 1;
     return isTopmostSeries ? [barRadius, 0] : [0, 0];
  };
}
```

that being said, i'm not a huge fan of what Prettier forces upon my original formatting :cry: :

```ts
  let barsBuilder = uPlot.paths.bars!({
    radius: (
      pctStacked ? 0 :
      !isStacked ? barRadius :
      (u: uPlot, seriesIdx: number) => {
          let isTopmostSeries = seriesIdx === u.data.length - 1;
          return isTopmostSeries ? [barRadius, 0] : [0, 0];
        }
    ),
```

Reply:
Yeah either way isn't super ideal, to me the if else just reads a bit more clearly but I don't feel strongly either way 🤷🏻‍♂️ maybe a helpful comment explaining basic logic flow could be useful, otherwise :shipit:

Reply:
:see_no_evil: